source/extensions/transport_sockets/tls/cert_validator/default_validator.cc

```diff
@@ -276,6 +303,86 @@ Envoy::Ssl::ClientValidationStatus DefaultCertValidator::verifyCertificate(
   return validated;
 }
 
+Ssl::ValidateResult DefaultCertValidator::doCustomVerifyCertChain(
```

Comment:
Since this method never returns Pending, is there any chance we could share more of the implementation with doVerifyCertChain?

Relatedly, should this only be called when the runtime flag is true, and vice versa? Will we eventually remove the non-async codepath?

Reply:
verifyCertAndUpdateStatus() is already shared. Only this method needs to instantiate X509_STORE_CTX. Though both of them call X509_verify_cert(), but the result handling are different. So the common code is very few lines, so I don't think it worth to extract them into helper function.

Eventually doVerifyCertChain() will be removed together with the runtime guard. And there is a TODO for that.

Reply:
Oh, gosh I misread the diff and didn't see the introduction of the new verifyCertAndUpdateStatus() method. Yes, you're right this code is already well shared. (And thanks for the TODO)